hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/listener/ClientSmartListenerService.java

```diff
@@ -200,6 +204,55 @@ public void run() {
         }, 1, 1, TimeUnit.SECONDS);
     }
 
+    private void openConnectionsToMembers() {
+        ClientClusterService clientClusterService = client.getClientClusterService();
+        long startTimeMillis = System.currentTimeMillis();
+        while (true) {
+            boolean isException = false;
+            Collection<Member> memberList = clientClusterService.getMemberList();
+            for (Member member : memberList) {
+                try {
+                    clientConnectionManager.getOrConnect(member.getAddress());
+                } catch (Exception e) {
+                    isException = true;
+                    if (!clientClusterService.getMemberList().contains(member)) {
+                        //member left the cluster no need to connect
+                        continue;
+                    }
+                    throwExceptionIfTimedOut(startTimeMillis, member, e);
+                }
+            }
+            if (!isException) {
+                return;
+            }
+
+            ClientInvocationServiceImpl invocationService = (ClientInvocationServiceImpl) client.getInvocationService();
+            try {
+                Thread.sleep(invocationService.getInvocationRetryPauseMillis());
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                throw ExceptionUtil.rethrow(e);
```

Comment:
should we rethrow or continue but also check in the loop for client lifecycle running flag, right?

Reply:
I've added isRunning check to while loop.